plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/TrinoGlueCatalog.java

```diff
@@ -339,15 +340,36 @@ public Table loadTable(ConnectorSession session, SchemaTableName table)
     @Override
     public void dropTable(ConnectorSession session, SchemaTableName schemaTableName)
     {
-        BaseTable table = (BaseTable) loadTable(session, schemaTableName);
-        validateTableCanBeDropped(table);
+        BaseTable table = null;
+        try {
+            table = (BaseTable) loadTable(session, schemaTableName);
+            validateTableCanBeDropped(table);
+        }
+        catch (Exception e) {
+            if (!(e.getCause() instanceof FileNotFoundException)) {
+                throw e;
+            }
+            LOG.warn("Failed to load table " + schemaTableName, e);
```

Comment:
Because we do not want to fail when `FileNotFoundException` exception occurs instead we want to drop the table from metastore and delete table location directory.